lms/services/email_preferences.py

```diff
@@ -29,8 +32,40 @@ class EmailPrefs:
     sat: bool = True
     sun: bool = True
 
-    def days(self) -> dict:
-        return {key: value for key, value in asdict(self).items() if key in self.DAYS}
+    @staticmethod
+    def user_preferences_key_for_email_digest_date(datetime_: datetime) -> str:
+        return f"instructor_email_digests.days.{EmailPreferences.DAYS[datetime_.weekday()]}"
+
+    def unsubscribe_instructor_digest(self) -> Self:
```

Comment:
Probably better if this method returns `None`: I think it's usual in Python (for clarity) for a method to _either_ return a new object (without mutating `self` or any of the arguments passed on) _or_ to mutate `self` but not both. Otherwise you end up with code that looks like it's just constructing a new object but is actually mutating the original object, which can lead to mistakes.